hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
 	}
 
 	private void checkTransactionNeeded() {
-		if ( !isTransactionInProgress() ) {
+		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
```

Comment:
If you already changed it it's fine.

On Wed, Jun 29, 2016, 7:33 AM Andrea Boriero <notifications@github.com>
wrote:

> In hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java
> <https://github.com/hibernate/hibernate-orm/pull/1459#discussion_r68937261>
> :
>
> > @@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
> >  	}
> >
> >  	private void checkTransactionNeeded() {
> > -		if ( !isTransactionInProgress() ) {
> > +		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
>
> i have just moved the code into the constructor and call the instance
> variable disallowOutOfTransactionUpdateOperations, it seems a little more
> readable not negation.
>
> @sebersole <https://github.com/sebersole> should I revert it or it's fine?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/hibernate/hibernate-orm/pull/1459/files/3a6add6c362a56509ff0a917412f85a87997300b..a7b5c9e7da94baf0393bd77c8db6316363d458a7#r68937261>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAOUE38X-tLh_GmiMGut7J8FJCEH0RhAks5qQmX3gaJpZM4JAEeF>
> .
>